packages/messageport/lib/messageport_tizen.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+const MethodChannel _channel = MethodChannel('tizen/messageport');
+
+/// Signature for a callback receiving message on messageport.
+///
+/// This is used by [LocalPort.register].
+typedef OnMessageReceived = Function(dynamic message, [RemotePort? remotePort]);
+
+/// Local port to receive messages
+class LocalPort {
+  LocalPort._(this._id, this._portName, this._isTrusted);
+
+  /// Register messageport and sets listener
+  ///
+  /// Names of already registered ports has to be unique.
+  /// Exception will be thrown when using on already registered port
+  void register(OnMessageReceived onMessage) {
+    if (_registeredPorts.contains(portName)) {
+      throw Exception('Port $portName is already registered');
```

Comment:
I made a small change which enables creating independent LocalPort object with the same name. In case of registering both (or more) local ports, all ports will receive the same message.